test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -182,6 +183,21 @@ private static void assertIndexUUIDs(BlobContainer repoRoot, RepositoryData repo
                 s -> s.startsWith("extra") == false).collect(Collectors.toList());
         }
         assertThat(foundIndexUUIDs, containsInAnyOrder(expectedIndexUUIDs.toArray(Strings.EMPTY_ARRAY)));
+        for (String indexId : foundIndexUUIDs) {
+            final Set<String> indexMetaGenerationsFound = indicesContainer.children().get(indexId)
+                .listBlobsByPrefix(BlobStoreRepository.METADATA_PREFIX).keySet().stream()
+                .map(p -> p.replace(BlobStoreRepository.METADATA_PREFIX, "").replace(".dat", ""))
+                .collect(Collectors.toSet());
+            final Set<String> indexMetaGenerationsExpected = new HashSet<>();
+            final IndexId idx =
+                repositoryData.getIndices().values().stream().filter(i -> i.getId().equals(indexId)).findFirst().get();
+            for (SnapshotId snapshotId : repositoryData.getSnapshots(idx)) {
+                indexMetaGenerationsExpected.add(repositoryData.indexMetaDataGenerations().indexMetaBlobId(snapshotId, idx));
+            }
+            // TODO: assertEquals(indexMetaGenerationsExpected, indexMetaGenerationsFound); requires cleanup functionality for
+            //       index meta generations blobs
+            assertTrue(indexMetaGenerationsFound.containsAll(indexMetaGenerationsExpected));
```

Comment:
Not checking equality here makes this test somewhat redundant admittedly but I figured it was ncie to leave the logic and TODO here to make it clear why we need index metadata blob cleanup.
Note, the fact that we leak index metadata blobs is not new and can happen without this change as well. As a matter of fact, you could argue that it is more likely to leak index metadata blobs before this change since there's simply more of them and the delete timing hasn't changed.

Reply:
Do we have a test that checks that index metadata is cleaned up if it is no longer referenced? (i.e. a succeeding snapshot delete that was holding the last reference to that index metadata)

Reply:
Enhanced the test for metadata deduplication in c9ab2fc6ddd to cover this explicitly